pkg/datagatherer/k8s/generic.go

```diff
@@ -110,3 +121,16 @@ func namespaceResourceInterface(iface dynamic.NamespaceableResourceInterface, na
 	}
 	return iface.Namespace(namespace)
 }
+
+// generateFieldSelector creates a field selector string from a list of
+// namespaces to exclude.
+func generateFieldSelector(excludeNamespaces []string) string {
+	var fieldSelector string
+	for _, excludeNamespace := range excludeNamespaces {
+		if excludeNamespace == "" {
+			continue
+		}
+		fieldSelector = fmt.Sprintf("%smetadata.namespace!=%s,", fieldSelector, excludeNamespace)
```

Comment:
This works for now, but you really should consider using `field.Selector` and constructing this in a safe way, instead of directly constructing a string to be sent as a selector to the server 😄

Reply:
Ah, didn't know that was a thing, will look into it!

Reply:
Have switched to using `fields.Selector` functions to build the selector. Am I correct in thinking it still needs to be converted to string with `.String()` to be used in the `ListOptions`?